cdptools/research_utils/transcripts.py

```diff
@@ -1,72 +1,154 @@
+        ],
+        order_by=(order_by_field, OrderOperators.desc),
+        limit=1
+    )
+
+    # Return result if found
+    if len(results) == 1:
+        return TrancriptJoin(event_id=event_id, transcript_details=results[0])
+
+    # Return none to be filtered out
+    return TrancriptJoin(event_id=event_id, transcript_details=None)
 
-def get_most_recent_transcript_manifest(db: Database) -> pd.DataFrame:
+
+def _get_file(file_id: str, db: Database) -> Dict[str, Any]:
+    return db.select_row_by_id(table="file", id=file_id)
+
+
+def get_transcript_manifest(db: Database, order_by_field: str = "confidence") -> pd.DataFrame:
     """
-    Get a pandas dataframe that can act as a manifest of the most recent transcript available for each event stored in a
-    CDP instance's database.
+    Get a pandas dataframe that can act as a manifest of a transcript available for each event stored in a CDP
+    instance's database.
 
     Parameters
     ----------
     db: Database
         An already initialized database object connected to a CDP instance's database.
+    order_by_field: str
+        Which field to order the transcripts by to select the first (highest value) of.
+        Default: "confidence"
+        Choices: ["created", "confidence"]
 
     Returns
     -------
     manifest: pandas.DataFrame
-        A dataframe with transcript, event, body, and file details where each row is the most recent transcript for the
-        event of that row.
+        A dataframe where each row has transcript, event, body, and file details for the event at that row.
     """
+
     # Get transcript dataset
-    transcripts = pd.DataFrame(db.select_rows_as_list("transcript"))
-    events = pd.DataFrame(db.select_rows_as_list("event"))
-    bodies = pd.DataFrame(db.select_rows_as_list("body"))
-    files = pd.DataFrame(db.select_rows_as_list("file"))
-    events = events.merge(bodies, left_on="body_id", right_on="body_id", suffixes=("_event", "_body"))
-    transcripts = transcripts.merge(files, left_on="file_id", right_on="file_id", suffixes=("_transcript", "_file"))
-    transcripts = transcripts.merge(events, left_on="event_id", right_on="event_id", suffixes=("_transcript", "_event"))
+    events = pd.DataFrame(db.select_rows_as_list("event", limit=1e6))
+
+    # Enforce that the provided order by field is valid
+    order_by_field = order_by_field.lower()
+    if order_by_field not in ALLOWED_ORDER_BY_FIELDS:
+        raise ValueError(
+            f"Provided `order_by_field` value is not a valid selection for transcript ordering / selection. "
+            f"Received: {order_by_field}. Possible choices: {ALLOWED_ORDER_BY_FIELDS}"
+        )
+
+    # Create transcript get partial
+    transcript_get = partial(_get_transcript_for_event, db=db, order_by_field=order_by_field)
+
+    # Threaded request all transcripts
+    with ThreadPoolExecutor() as exe:
+        transcript_joins = list(exe.map(transcript_get, list(events.event_id)))
```

Comment:
I like the creation of the TranscriptJoin class that simplifies the code and allows ThreadPoolExecutor use, I probably would've tried an overcomplicated db query.